src/nodes.rs

```diff
@@ -609,6 +610,7 @@ impl Ast {
             open: true,
             last_line_blank: false,
             table_visited: false,
+            line_offsets: Vec::with_capacity(0),
```

Comment:
I changed to using `with_capacity(0)` in the second commit. The idea being there many ast nodes that will never use `line_offsets`, so why keep around default allocated storage for them.

Maybe it's not an issue. Or maybe it would be better to make it an Option that gets allocated only when we need to add something it. 🤔

Reply:
I agree, I wasn't relishing the thought of using an Option. Just wasn't sure if this was Rust-ish enough. 👍